ansible/roles/test/files/ptftests/device_connection.py

```diff
@@ -62,18 +63,17 @@ def execCommand(self, cmd, timeout=DEFAULT_CMD_EXECUTION_TIMEOUT_SEC):
         retValue = 1
         try:
             client.connect(self.hostname, username=self.username,
-                           password=self.password, allow_agent=False)
+                           password=self.passwords[self.password_index], allow_agent=False)
             si, so, se = client.exec_command(cmd, timeout=timeout)
             stdOut = so.readlines()
             stdErr = se.readlines()
             retValue = 0
         except AuthenticationException as authenticationException:
             logger.error('SSH Authentication failure with message: %s' %
                          authenticationException)
-            if self.alt_password is not None and self.alt_password_index < len(self.alt_password):
-                # attempt retry with alt_password
-                self.password = self.alt_password[self.alt_password_index]
-                self.alt_password_index += 1
+            if len(self.passwords) > 1:
+                # attempt retry with another password
+                self.password_index = (self.password_index + 1) % len(self.passwords)
```

Comment:
Where will this be retried from and what's the limit of retry?

Reply:
There's a `@retry` decorator for this function that retries 4 times upon getting an `AuthenticationException`

Reply:
Ideally yes, but these variables are not defined when the `@retry` decorator is evaluated, so it can't be used. It actually might be a bit easier to do this without using the `@retry` decorator and doing the retry logic ourselves.